src/cpu/dyn_cache.h

```diff
@@ -125,7 +126,7 @@ static uint8_t *cache_code_start_ptr = nullptr;
 static uint8_t *cache_code = nullptr;
 static uint8_t *cache_code_link_blocks = nullptr;
 
-static CacheBlock *cache_blocks = nullptr;
+static std::array<CacheBlock, CACHE_BLOCKS> cache_blocks;
```

Comment:
Suggest value-initializing the array w/ `= {};`, just as a good rule of thumb. 

Painful descriptions in language land here: https://stackoverflow.com/questions/18295302/default-initialization-of-stdarray

We know the Cache block class doesn't use default values for its members and doesn't have a user-provided  constructor, so the last hope to zero them all is with {} (or perhaps a memcpy).